main.go

```diff
@@ -146,40 +145,46 @@ func handle(w http.ResponseWriter, r *http.Request) {
 			http.Error(w, err.Error(), http.StatusInternalServerError)
 			return
 		} else if errors.Is(err, fs.ErrNotExist) {
-			// go build
-			args := []string{"build", "-o", filepath.Join(output, "main.wasm")}
+			// go run
+			args := []string{"run", "-exec", "cat"}
 			if *flagTags != "" {
 				args = append(args, "-tags", *flagTags)
 			}
 			if *flagOverlay != "" {
 				args = append(args, "-overlay", *flagOverlay)
 			}
-			if len(flag.Args()) > 0 {
+			if flag.NArg() > 0 {
 				args = append(args, flag.Args()[0])
 			} else {
 				args = append(args, ".")
 			}
+
 			log.Print("go ", strings.Join(args, " "))
+
 			cmdBuild := exec.Command("go", args...)
 			cmdBuild.Env = append(os.Environ(), "GOOS=js", "GOARCH=wasm")
-			cmdBuild.Dir = workdir
+			cmdBuild.Dir = "."
 			out, err := cmdBuild.CombinedOutput()
 			if err != nil {
 				log.Print(err)
 				log.Print(string(out))
 				http.Error(w, string(out), http.StatusInternalServerError)
 				return
 			}
-			if len(out) > 0 {
-				log.Print(string(out))
-			}
 
-			f, err := os.Open(filepath.Join(output, "main.wasm"))
+			f, err := os.Create(filepath.Join(output, "main.wasm"))
 			if err != nil {
 				http.Error(w, err.Error(), http.StatusInternalServerError)
 				return
 			}
 			defer f.Close()
+
+			_, err = f.Write(out)
+			if err != nil {
```

Comment:
`if _, err := f.Write(out); err != nil {`

Reply:
resolving since we're using cp instead of write now